proxy/src/config.rs

```diff
@@ -36,7 +36,24 @@ pub struct ProxyConfig {
     pub auth_link_uri: ApiUrl,
 }
 
-pub type TlsConfig = Arc<rustls::ServerConfig>;
+#[derive(Clone)]
+pub struct TlsConfig {
+    pub tls_config: Arc<rustls::ServerConfig>,
+    pub common_name: Option<&'static str>,
```

Comment:
I used 'Option<&'static str>' and not '&'static str', because there are some tests that construct this struct using TlsConfig::new, where common_name is set None (because they are isolated tests that don't create a common name).

-----

ok, I'll implement the second suggestion.